src/three-domain-secure/component.jsx

```diff
@@ -34,21 +34,34 @@ const parseSdkConfig = ({ sdkConfig, logger }): SdkConfig => {
   return sdkConfig;
 };
 
-const parseMerchantPayload = ({
+export const parseMerchantPayload = ({
   merchantPayload,
 }: {|
   merchantPayload: MerchantPayloadData,
 |}): requestData => {
-  const { threeDSRequested, amount, currency, nonce, transactionContext } =
-    merchantPayload;
+  const {
+    threeDSRequested,
+    threeDSTriggerMode,
+    amount,
+    currency,
+    nonce,
+    transactionContext,
+  } = merchantPayload;
+
+  let verificationMethod = "SCA_WHEN_REQUIRED";
+
+  if (threeDSRequested !== undefined) {
+    verificationMethod = threeDSRequested ? "SCA_ALWAYS" : "SCA_WHEN_REQUIRED";
+  } else if (threeDSTriggerMode) {
+    verificationMethod = threeDSTriggerMode;
+  }
```

Comment:
Checking `threeDSRequested` before `threeDSTriggerMode` is making the `should prioritize threeDSTriggerMode over threeDSRequested` test fail. Changing the order switches the priority to `threeDSTriggerMode` and makes the test pass. Would something like this work with the rest of the implementatino?

```suggestion
  if (threeDSTriggerMode) {
    verificationMethod = threeDSTriggerMode;
  } else if (threeDSRequested !== undefined) {
    verificationMethod = threeDSRequested ? "SCA_ALWAYS" : "SCA_WHEN_REQUIRED";
  } 
```

Reply:
Actually that is what @jtanya17  had originally. I'll revert my last commit.